### PR TITLE
test(python): Reorganize benchmark test folder

### PR DESCRIPTION
**Pull request by stinodego**
Related to #6364

Changes:
* Added a threshold for the H2O AI benchmark - strings test.
  * I figured, if there is no threshold set, there is not much use in running the benchmark at all.
* Made other benchmark tests into proper `pytest` tests and marked them as `benchmark` tests.
  * These are now ran as part of the `benchmark` workflow, and their run durations are reported in the logs.
* Rename the folder `db-benchmark` to `benchmark` and add some documentation to the benchmark script.
* Rename `docs/run_doc_examples.py` to `docs/run_doctest.py`
* Added a `README.md` to the `py-polars/tests` folder, and linked to this in the `CONTRIBUTING` guide. Feels good to make a bunch of this knowledge explicit to future contributors :relaxed:

### CODE REVIEW

File: py-polars/tests/unit/test_series.py [hunk 1]
```diff
@@ -2055,11 +2055,6 @@ def test_duration_arithmetic() -> None:
         assert_series_equal(df1["d_offset"], df2["d_offset"])
 
 
-def test_mean_overflow() -> None:
-    arr = np.array([255] * (1 << 17), dtype="int16")
-    assert arr.mean() == 255.0
```
Comment (stinodego) [inline py-polars/tests/unit/test_series.py:2060]: This doesn't test anything Polars-related, so I removed it.

Comment (ritchie46) [reply py-polars/tests/unit/test_series.py:2060]: It does! It was testing if it doesn't overflow the mean. A naive sum/count would overflow the i16

**Review by ritchie46 — COMMENTED** [review]
The slow tests were explicitly run in benchmark folder because the polars had a release build. So we can test really slow code.

**Comment by stinodego** [comment]
Ah interesting, learned a lot from your comments 😄 I'm gonna revert a lot of this stuff and add a bit of documentation in that folder on what's going on there.

File: py-polars/tests/unit/test_series.py (hunk 1 above)
Comment (stinodego) [reply py-polars/tests/unit/test_series.py]: Hmm, but it's just `numpy` code though? 🤔 We don't rely on this to calculate our mean, right?

Reverted this removal, but I still don't understand why this test is in there!

**Comment by stinodego** [comment]
> Yeah, maybe most ideal would be having those loose files also concerted to pytest functions and running those "release-build" tests only in CI once the benchmark has finished. That same binary is then used for those tests.

Done, and documented in the new Test Suite README 😄

File: py-polars/tests/benchmark/test_release.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
+"""
```
Comment (ritchie46) [inline py-polars/tests/benchmark/test_release.py:1]: Nice, that's cleaner indeed. :)

**Review by ritchie46 — APPROVED** [review]
Nice cleanup. Especially formalizing the release tests is a good one. 

I hope we can add the full TPCH benchmarks in the future.

**Comment by alexander-beedie** [comment]
Really nice.

**Pull request merged**